### ISSUE TITLE
expose the current database type to Up/Down

### ISSUE THREAD

**Issue by aarondandy** [fluentmigrator/fluentmigrator#574]
Would it be possible to expose the current database type to the migration? This would allow me to write logic in C# to perform migrations in different ways depending on the database I was using. IfDatabase is OK but requires a lot of duplicate code if more than two databases are used. For example many parts of my migrations may be the same for postgres and sqlite but would be different only for oracle. Having something like an ElseDatabase would have been nice but it would have been really nice and simple if I could just write `if(DatabaseType=="oracle"){}else{}` . My tale does have a happy ending though as we just ended up moving to postgres finally, so maybe this is all unneeded, but exposing just a string would have helped a lot during that transition and with tests.

**Comment by dealproc** [comment]
:+1: This would be a valuable way to build a migration, I believe.

**Comment by dealproc** [comment]
Are we able to utilize Profiles somehow to achieve this?  My initial thought was maybe we could create multiple `[Profile()]` attributes on a migration, but according to the code here: https://github.com/schambers/fluentmigrator/blob/b76fb3c7c25f7ab455cd940d2cb560058892813e/src/FluentMigrator/ProfileAttribute.cs it looks like we're not able to do this.

Does the concept of a profile allow something like `[Profile("oracle,sqlserver,postgres")]` ?

**Comment by tommarien** [comment]
Hi,

Like already discussed previously see #387, the ifdatabase should take a predicate. This would solve a lot of these issues. But it should be kept in a backwards compatible manner.

Regards

**Comment by aarondandy** [comment]
@tommarien 

So it would be something like this?

```
Action<T> normalWay = (stuff) => ...;

IfDatabase("oracle").SomeSpecialOracleStuff();
IfDatabase("postgres").Do(normalWay);
IfDatabase("sqlite").Do(normalWay);
```

That could work to reduce some redundancy.

**Comment by tommarien** [comment]
@aarondandy No the clause itself should allow a predicate

Like IfDatabase(x=>x == "oracle" || x== "Postgres")

**Comment by dealproc** [comment]
I like @tommarien 's way better.  The only other thing I may ask/request... can we enumerate the types instead of depending on magic strings?

**Comment by detoxhby** [comment]
👍 #422

**Comment by fubar-coder** [comment]
Fixed by #683
